### PR TITLE
Arn 462 integration assessments api

### PR DESCRIPTION
**Pull request by adelaidaa**
This is still WIP for the integration with the new endpoints in `offender-assessment-api` but i want see if i can get some E2E flow already working and then use the responses from `offender-assessment-api` to bundle all the data for the new endpoints.

some tests are still missing but coming up in the next PR's

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/RiskService.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+@Service
+class RiskService(private val assessmentClient: AssessmentApiRestClient) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getRoshRisksByCrn(crn: String): AllRoshRiskDto {
+    log.info("Get Rosh Risk for crn $crn")
+    val sectionsAnswers = assessmentClient.getRoshSectionsForCompletedLastYearAssessment(crn)
+    log.info("Section answers for crn $crn $sectionsAnswers")
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/RiskService.kt:24]: for now just print results from offender-assessments-api just to see if it's all correctly configured

File: src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/RiskService.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      RiskRoshSummaryDto(
+        "whoisAtRisk",
+        "natureOfRisk",
+        "riskImminence",
+        "riskIncreaseFactors",
+        "riskMitigationFactors",
+        mapOf(RiskLevel.HIGH to listOf("children")),
+        mapOf(RiskLevel.MEDIUM to listOf("known adult"))
+      )
+    )
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/RiskService.kt:47]: will use results for OFAA soon to fill this hardcoded dtos

File: src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/exceptions/ApplicationsExceptions.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.http.HttpMethod
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.SupplementaryRiskDto
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.restclient.ExternalService
+
+class EntityNotFoundException(msg: String?) : RuntimeException(msg)
+class DuplicateSourceRecordFound(msg: String?, val supplementaryRiskDto: SupplementaryRiskDto? = null) :
+  RuntimeException(msg)
+
+// External Services Exceptions
+class ExternalApiEntityNotFoundException(
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/exceptions/ApplicationsExceptions.kt:12]: need to add controller advice for this new exceptions also

File: src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/RiskService.kt (hunk 2 above)
Comment (LucasCairns) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/RiskService.kt:47]: 👍

**Review by LucasCairns — APPROVED** [review]
LGTM :shipit:

**Pull request merged**